simulator/func_sim/rf/rf.h

```diff
@@ -30,13 +30,19 @@ class RF
         return (accumulation == +1 && MAX_VAL32 - x < y) || (accumulation == -1 && x < y) ? 1 : 0;
     }
 
-protected:
+public:
+    RF() = default;
 
     const auto& read( Register num) const
     {
         return get_value( num);
     }
 
+    const auto& read( size_t regno) const
+    {
+        return array.at( regno);
```

Comment:
I thinked about `return read (Register (regno))` here, however, I think converting number to `Register` and then converting it back in `get_value()` is a waste of resources.

Reply:
And, as a result, MIPS decoder uses `from_cpu_index` and `from_cp0_index`, GDB uses `from_gdb_index`, RF would use `to_rf_index` — and nobody care about the real index inside

Reply:
Looks nice. I'll implement it.

Isn't calling a class constructor, then calling one method and then calling another method is slower than not calling them at all?

Reply:
> calling class constructor, then calling one method and then calling another method is slower than not calling them at all?

Let me do measurements for you: https://godbolt.org/z/F95VSg.

Anyway, procedure calls are not the most valuable resource being compared to the time of the developer.